tests/e2e_tests/integration/07-saml.spec.ts

```diff
@@ -0,0 +1,183 @@
+import axios from 'axios';
+import * as fs from 'fs';
+import * as https from 'https';
+import jwtDecode from 'jwt-decode';
+
+import test, { expect } from '../fixtures/fixtures';
+
+test.describe('SAML Login', () => {
+  test.use({ storageState: 'storage.json' });
+  test.describe('SAML login via sso/id/login', () => {
+    test('Logs in using UI', async ({ environment, context, baseUrl, page, password, username }) => {
+      test.skip(environment !== 'staging');
+      test.setTimeout(240000);
+      const saml_settings={
+        'login':'morty',
+        'password':'panic',
+        'email':'msmith@samltest.id',
+        'idp_url':'https://samltest.id/saml/idp'
+      };
+      const timeout_4s=4000;
+      const timeout_1s=1000;
+      const storage=await context.storageState();
+      const jwt=storage['cookies'].find(cookie => cookie.name === 'JWT').value;
+
+      const idp_response = await axios({
+        url: saml_settings['idp_url'],
+        method: 'GET'
+      });
+
+      if (idp_response.status !== 200) {
+        throw 'expected 201';
+      }
+
+      const metadata = idp_response.data;
+
+      console.log(`running with ${baseUrl}`);
+      console.log(`IdP metadata len=${metadata.length} loaded and uploading`);
+
+      await page.goto(`${baseUrl}ui/settings/organization-and-billing`);
+
+      await expect(page).toHaveURL(`${baseUrl}ui/settings/organization-and-billing`);
```

Comment:
just a sidenote: while I can see the benefit of not having to worry about which calls might return async - most of your `expect` calls don't expect async results, so `await`ing them is most likely not needed...

Reply:
I believe toHaveURL needs await:

```js
  /**
   * Ensures the page is navigated to the given URL.
   *
   * ```js
   * await expect(page).toHaveURL(/.*checkout/);
   * ```
   *
   * @param urlOrRegExp Expected URL string or RegExp.
   * @param options
   */
  toHaveURL(urlOrRegExp: string|RegExp, options?: {
    /**
     * Time to retry the assertion for. Defaults to `timeout` in `TestConfig.expect`.
     */
    timeout?: number;
  }): Promise<void>;
```

Reply:
that's why I wrote _most of your `expect` calls_ 😉 - OTOH since you `.goto()` in the line before you can likely remove the assertion entirely.